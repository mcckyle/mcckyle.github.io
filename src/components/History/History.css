/* Filename: History.css */

/* ====================================
   History Styles.
======================================= */
#history .section-title
{
    color: var(--highlight);
}

.history-list
{
    list-style: none;
    padding: 0;

    margin-top: 2rem;
    display: grid;
    gap: 2rem;
}

/* History Timeline Item Styles. */
.history-item
{
    display: flex;
    align-items: flex-start;
    gap: 1.25rem;
    position: relative;

    background: var(--surface);
    padding: 1.5rem 1.75rem;
    border-radius: 1rem;
    box-shadow: 0 6px 18px rgba(0, 0, 0, 0.05);

    transition: transform var(--transition), box-shadow var(--transition);
}

.history-item:hover
{
    transform: translateY(-3px);
    box-shadow: 0 10px 28px rgba(0, 0, 0, 0.08);
}

/* History Timeline Line Styles. */
/*.history-item::before
{
    content: "";
    position: absolute;
    left: 20px;
    top: 0;
    bottom: 0;
    width: 2px;
    background: var(--highlight);
    opacity: 0.25;
}*/

/* History Timeline Marker Styles. */
.history-marker
{
    flex-shrink: 0;
    width: 14px;
    height: 14px;
    background: var(--highlight);
    border-radius: 50%;
    margin-top: 0.5rem;
    box-shadow: 0 0 0 3px var(--surface);
}

.history-role
{
    font-size: 1.25rem;
    font-weight: 600;
    color: var(--primary);
    margin-bottom: 0.35rem;

    transition: color var(--transition);
}

.history-role:hover
{
    color: var(--accent);
}

.history-company
{
    font-size: 1rem;
    color: var(--accent);
    margin-bottom: 0.25rem;
}

.history-period
{
    font-size: 0.9rem;
    color: var(--text-muted);
}
