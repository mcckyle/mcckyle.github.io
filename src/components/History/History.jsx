//Filename: History.jsx
//Author: Kyle McColgan
//Date: 29 September 2025
//Description: This file contains the component for the React personal project professional history section.

import React from 'react';
import './History.css';

function History()
{
    const experiences = [
      {
        company: "Primary Systems",
        role: "IT System Engineer",
        period: "May 2023 - September 2023"
      },
      {
          company: "Walmart",
          role: "Software Engineer",
          period: "July 2021 - October 2022"
      }
    ];

    return (
        <section id="history" className="history" aria-label="Professional History">
          <div className="history-container">
            <h2 className="history-title">Professional History</h2>
            <ul className="history-list">
            {experiences.map((exp, index) => (
              <li key={index} className="history-item">
                <article>
                  <h3 className="history-role">{exp.role}</h3>
                  <p className="history-company">{exp.company}</p>
                  <time className="history-period">{exp.period}</time>
                </article>
              </li>
            ))}
            </ul>
          </div>
        </section>
    );
}

export default History;
