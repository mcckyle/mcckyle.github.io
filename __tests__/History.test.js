//Filename: History.test.js
//Author: Kyle McColgan
//Date: 28 October 2025
//Description: This file contains the Jest unit tests for the personal React project history component.

import React from 'react';
import { render, screen, within } from '@testing-library/react';
import '@testing-library/jest-dom';
import History from '../src/components/History/History.jsx';

describe('History component', () => {
    beforeEach(() => {
        render(<History />);
    });

    //Test #1
    it('renders the History section', () => {
        const div = screen.getByRole('region', { name: /Professional History/i });
        expect(div).toBeInTheDocument();
    });

    //Test #2
    it('renders the section title correctly', () => {
        const title = screen.getByRole('heading', { level: 2 });
        expect(title).toHaveTextContent('Professional History');
    });

    //Test #3
    it('renders the list of job experiences', () => {
        const listItems = screen.getAllByRole('listitem');
        expect(listItems).toHaveLength(2);
    });

    //Test #4
    it('renders the first job experience correctly', () => {
        const firstItem = screen.getAllByRole('listitem')[0];
        expect(within(firstItem).getByText(/IT System Engineer/i)).toBeInTheDocument();
        expect(within(firstItem).getByText(/Primary Systems/i)).toBeInTheDocument();
        expect(within(firstItem).getByText(/May 2023 - September 2023/i)).toBeInTheDocument();
    });

    //Test #5
    it('renders the second job experience correctly', () => {
        const secondItem = screen.getAllByRole('listitem')[1];
        expect(within(secondItem).getByText(/Software Engineer/i)).toBeInTheDocument();
        expect(within(secondItem).getByText(/Walmart/i)).toBeInTheDocument();
        expect(within(secondItem).getByText(/July 2021 - October 2022/i)).toBeInTheDocument();
    });

    //Test #6
    it('has the correct ARIA role of "region"', () => {
        const div = screen.getByRole('region', { name: /Professional History/i });
        expect(div).toBeInTheDocument();
    });

    //Test #7
    it('ensures no empty list items', () => {
        const listItems = screen.getAllByRole('listitem');
        listItems.forEach(item => {
            expect(item).not.toBeEmptyDOMElement();
        });
    });

    //Test #8
    it('renders job titles in the list items', () => {
        expect(screen.getByText(/IT System Engineer/i)).toBeInTheDocument();
        expect(screen.getByText(/Software Engineer/i)).toBeInTheDocument();
    });

    //Test #9
    it('does not log any errors to the console', () => {
        const consoleError = jest.spyOn(console, 'error').mockImplementation();
        expect(consoleError).not.toHaveBeenCalled();
        consoleError.mockRestore();
    });

    //Test #10
    it('matches the snapshot of the History section', () => {
        const { asFragment } = render(<History />);
        expect(asFragment()).toMatchSnapshot();
    });
});
