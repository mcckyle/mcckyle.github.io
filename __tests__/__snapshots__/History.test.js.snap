// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component matches the snapshot of the History section 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="history-title"
    class="section-content"
    id="history"
    role="region"
  >
    <h2
      class="section-title"
      id="history-title"
    >
      Professional History
    </h2>
    <ul
      class="history-list"
    >
      <li
        class="history-item"
      >
        <div
          aria-hidden="true"
          class="history-marker"
        />
        <div
          class="history-content"
        >
          <h3
            class="history-role"
          >
            IT System Engineer
          </h3>
          <p
            class="history-company"
          >
            Primary Systems
          </p>
          <time
            class="history-period"
          >
            May 2023 - September 2023
          </time>
        </div>
      </li>
      <li
        class="history-item"
      >
        <div
          aria-hidden="true"
          class="history-marker"
        />
        <div
          class="history-content"
        >
          <h3
            class="history-role"
          >
            Software Engineer
          </h3>
          <p
            class="history-company"
          >
            Walmart
          </p>
          <time
            class="history-period"
          >
            July 2021 - October 2022
          </time>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
